### PR TITLE
Remove spring-platform

### PR DESCRIPTION
**Pull request by erikkemperman**
Don't merge this yet, couchbase sample failing. See https://github.com/jhipster/generator-jhipster/issues/7422#issuecomment-379789335.

### CODE REVIEW

File: jhipster-server/pom.xml [hunk 1]
```diff
@@ -213,17 +213,6 @@
 
     <build>
         <plugins>
-            <plugin>
-                <groupId>org.sonatype.plugins</groupId>
```
Comment (jdubois) [inline jhipster-server/pom.xml]: are you sure this is not needed anymore? I think I need this to publish snapshot releases

Comment (erikkemperman) [reply jhipster-server/pom.xml]: The same config is in the parent:
https://github.com/jhipster/jhipster/blob/master/jhipster-dependencies/pom.xml#L839
And we just need to enable the plugin:
https://github.com/jhipster/jhipster/pull/55/files#diff-580af11172ef8dd3e17b50bdf019f094L250

**Comment by jdubois** [comment]
@erikkemperman the issue you have with Couchbase is a NoSuchMethodError on the the Couchbase client lib - so that's just a versioning issue

**Comment by erikkemperman** [comment]
Yeah, I saw that, but as far as I can tell there are only patch-level increments with this change (I diffed the output of `mvn dependency:tree` before and after) and I would expect a disappearing method to be a major change!

**Comment by erikkemperman** [comment]
The problematic change is here:
https://github.com/couchbase/couchbase-java-client/commit/762c2dc4460720fc5d27719630694e8badc9cd10#diff-5c321e9c5d4bec4a286000165da7bbd8L293

I have a lot of other things to do today, but will try to find some time to investigate further!

**Comment by erikkemperman** [comment]
@jdubois Fixed!

**Comment by jdubois** [comment]
Awesome, let's do this!

**Comment by jdubois** [comment]
Let's test a little bit, and then I'll deploy a new snapshot - all JHipster v5 users will get it tomorrow, as it's a snapshot :-)

**Pull request merged**